src/kernel/mconfig_dot.ml

```diff
@@ -220,213 +74,154 @@ let parse_suffix str =
+      let stderr = Unix.in_channel_of_descr stderr_r in
+      { pid; stdin; stdout; stderr }
+  end
+
+  let running_processes : (string * t, Process.t) Hashtbl.t = Hashtbl.create 0
+
+  let get_process ~dir configurator =
+    try
+      let p = Hashtbl.find running_processes (dir, configurator) in
+      let i, _ = Unix.waitpid [ WNOHANG ] p.pid in
+      if i = 0 then
+        p
+      else
+        let p = Process.start ~dir configurator in
+        Hashtbl.replace running_processes (dir, configurator) p;
+        p
+    with Not_found ->
+      let p = Process.start ~dir configurator in
+      Hashtbl.add running_processes (dir, configurator) p;
+      p
+end
+
 let postprocess_config config =
   let clean list = List.rev (List.filter_dup list) in
   {
-    dot_merlins  = config.dot_merlins;
     build_path   = clean config.build_path;
     source_path  = clean config.source_path;
     cmi_path     = clean config.cmi_path;
     cmt_path     = clean config.cmt_path;
-    packages     = clean config.packages;
     extensions   = clean config.extensions;
     suffixes     = clean config.suffixes;
     flags        = clean config.flags;
-    findlib_path = clean config.findlib_path;
     stdlib      = config.stdlib;
-    findlib     = config.findlib;
     reader      = config.reader;
-    findlib_toolchain = config.findlib_toolchain;
     exclude_query_dir = config.exclude_query_dir;
   }
 
-
-let load ~stdlib filenames =
-  let filenames = List.map ~f:canonicalize_filename filenames in
-  let filenames = List.filter_map ~f:find filenames in
-  let directives = directives_of_files filenames in
-  let config =
-    List.fold_left directives ~init:empty_config
-      ~f:(fun config file -> prepend_config ~stdlib file config)
-  in
-  postprocess_config config
-
-(* FIXME: Move elsewhere, processing of findlib packages*)
-
-let ppx_of_package ?(predicates=[]) setup pkg =
-  let d = Findlib.package_directory pkg in
-  (* Determine the 'ppx' property: *)
-  let in_words ~comma s =
-    (* splits s in words separated by commas and/or whitespace *)
-    let l = String.length s in
-    let rec split i j =
-      if j < l then
-        match s.[j] with
-        | (' '|'\t'|'\n'|'\r'|',' as c) when c <> ',' || comma ->
-          if i<j then (String.sub s ~pos:i ~len:(j-i)) :: (split (j+1) (j+1))
-          else split (j+1) (j+1)
-        |	_ ->
-          split i (j+1)
-      else
-      if i<j then [ String.sub s ~pos:i ~len:(j-i) ] else []
-    in
-    split 0 0
-  in
-  let resolve_path = Findlib.resolve_path ~base:d ~explicit:true in
-  let ppx =
-    try Some(resolve_path (Findlib.package_property predicates pkg "ppx"))
-    with Not_found -> None
-  and ppxopts =
+type context = string * Configurator.t
+
+let get_config (dir, cfg) path =
+  try
+    let p = Configurator.get_process ~dir cfg in
+    let cmd = Dot_protocol.Commands.make_f path in
+    (* TODO: ensure [path] is absolute, or that it is relative to dir, and not the cwd. *)
```

Comment:
@trefis The only call to that function is in [Mconfig](https://github.com/ocaml/merlin/pull/1123/files#diff-4fa8910acf164e080175042c6ad98d87R212) and is preceded by a call to `canonicalize_filename`. Thus it should always be absolute, shouldn't it ?

Reply:
Yes